app/js/enhancedDate.js

```diff
@@ -1,5 +1,159 @@
+ *
+ * getMonthName: Return the full month name as a string, e.g. "January", "February", etc.
+ *
+ * isFuture: Return boolean true if date is in the future (from when method is called)
+ *
+ * isToday: Return boolean true if the stored date is "today".
+ */
 
+var enhancedDate = (function () {
+    // private data variables
+    var myDate;
+    
+    return {
+        // public methods and properties
+        
+        setDate: function(dateItem) {
+            console.log("setDate");
+            if (dateItem === undefined) {
+                myDate = new Date();
+            } else {
+                myDate = new Date(dateItem);
+            }
+        },
+        
+        getDate: function(dateItem) {
+            console.log("getDate");
+            if (dateItem) {
+                return myDate;
+            } else {
+                return Date.UTC(myDate);
+            }
+
+        },
+        
+        getDayName: function() {
+            console.log("getDayName");
+            var day = myDate.getDay();
+            switch (day) {
+                case 0:
+                    return "Sunday";
+                case 1:
+                    return "Monday";
+                case 2:
+                    return "Tuesday";
+                case 3:
+                    return "Wednesday";
+                case 4:
+                    return "Thursday";
+                case 5:
+                    return "Friday";
+                case 6:
+                    return "Saturday";
+            }
+        },
+        
+        getMonthName: function() {
+            console.log("getMonthName");
+            var month = myDate.getMonth();
+            switch (month) {
+                case 0:
+                    return "January";
+                case 1:
+                    return "February";
+                case 2:
+                    return "March";
+                case 3:
+                    return "April";
+                case 4:
+                    return "May";
+                case 5:
+                    return "June";
+                case 6:
+                    return "July";
+                case 7:
+                    return "August";
+                case 8:
+                    return "September";
+                case 9:
+                    return "October";
+                case 10:
+                    return "November";
+                case 11:
+                    return "December";
+            }
+        },
+        
+        isFuture: function() {
+            console.log("isFuture");
+            var now = Date.now();
+            if (myDate > now) {
+                return true;
+            } else {
+                return false;
+            }
+        },
+        
+        isToday: function() {
+            console.log("isToday");
+            var now = Date.now();
+            if (this.getDate(myDate) == this.getDate(now)) {
```

Comment:
When you call this.getDate( ) with a truthy value (both myDate and now will evaluate to truthy in this case), it will simply return myDate as a Date object. This means you're comparing if myDate == myDate. What you want to compare is if myDate's year, month, and date are the same as now's year, month, and date. To get "now", just do `var now = new Date( );`